src/lib/stitching/exchange/stitching.ts

```diff
@@ -351,6 +352,90 @@ export const exchangeStitchingEnvironment = ({
           },
         },
       },
+      Mutation: {
+        // monkey-patch the existing exchange mutation
+        // to add operations dependent on impulse
+        commerceCreateInquiryOfferOrderWithArtwork: {
+          resolve: async (_source, args, context, info) => {
+            const {
+              conversationLoader,
+              conversationCreateConversationOrderLoader,
+            } = context
+            const {
+              input: { impulseConversationId },
+            } = args
+
+            try {
+              await conversationLoader(impulseConversationId)
```

Comment:
This is interesting. We are talking to both Exchange and Impulse, and before we create an Exchange order with a given Impulse conversation id, we want to verify if it's legit (e.g. conversation exists, user is authorized to access it, etc.).

I'm thinking verifying this kind of external id can be optional and we can leave it to the actual service (i.e. Impulse in this case) to verify eventually. For example, if an invalid Impulse conversation id is provided, the [create conversation order loader](https://github.com/artsy/metaphysics/pull/2974/files#diff-91e52543c599d3054ca57ac90b030871b4bc5a2ab9250fb8a39cda0c7f9e3585R424) below should fail. We can still be extra cautious here and monitor the performance.

Reply:
@starsirius we just wanted to make sure we verify the validity of the conversation ID before creating the exchange order (`deligateToSchema`  function below). It is not the end of the world if we create a `Pending` order in exchange with a wrong conversation `id` because pending orders don't mean much as long they are not submitted. This just ensures all conversation ids in exhange are valid.

Reply:
Our case here would be to validate that the 'conversation id' is something more than just an arbitrary string. Either exchange could do it (call impulse) or metaphysics could do it, which gives us the additional user access check before any data is changed.

Reply:
Yeah I'm fine being extra cautious here! 👍  I was mainly thinking it's hard to control external records (e.g. impulse conversations can be deleted in the future) so a broken link may not be that bad as long as the external service does proper check like authorization. And in our case, we cannot roll back the exchange order just created but it would eventually expire and buyers have to start a new one (after we do [this](https://artsyproduct.atlassian.net/browse/PURCHASE-2354)).